src/gc/impl/conservative/gc.d

```diff
@@ -2078,6 +2078,135 @@ struct Gcx
     ToScanStack!(ScanRange) toscanPrecise;
 
     /**
+     * Finds the offset between an index and the next set bit, up to our max.
+     *
+     * Returns: The offset between index and the next bit,
+     *          or -1 if no bits are set between index and max
+     */
+    int bitScanForwardOffset(size_t* bitArray, size_t index, size_t max) nothrow
```

Comment:
I'm in doubt that this function is currently very efficient:
- I would not trust the DMD optimizer that recalculation of bitIndex and testBit is efficient
- using floating point log2 is unlikely to be fast
- bsf is undefined for argument 0, this is not Windows specific
- bitScanForwardOffset should not be recursive
- core.bitop.bsf(shifted) is called twice